derive-impl/src/util.rs

```diff
@@ -178,6 +178,27 @@ impl ItemMetaInner {
         Ok(value)
     }
 
+    /// key exist, no str value -> Ok(Some(None))
+    ///
+    /// key exist, have str value -> Ok(Some(Some(str)))
+    ///
+    /// key doesn't exist -> Ok(None)
+    pub fn _optional_key_with_optional_str(&self, key: &str) -> Result<Option<Option<String>>> {
+        if let Some((_, meta)) = self.meta_map.get(key) {
+            let inner = match meta {
+                Meta::NameValue(syn::MetaNameValue {
+                    lit: syn::Lit::Str(lit),
+                    ..
+                }) => Some(lit.value()),
+                Meta::Path(_) => None,
+                _ => bail_span!(meta, r#"Expect `trace` or `trace = "manual""#),
+            };
+            Ok(Some(inner))
+        } else {
+            Ok(None)
+        }
```

Comment:
The new syntax help to reduce indent.
btw, I think this helper is not general to reuse from other place.
combination of `_has_key` and `_optional_str` will be more useful and easy to understand.
(`Option<Option<T>>` is not very human friendly and it doesn't give more information than `_has_key` and `_optional_str`)

```suggestion
        let Some((_, meta)) = self.meta_map.get(key) else {
            return Ok(None);
        };
        let inner = match meta {
            Meta::NameValue(syn::MetaNameValue {
                lit: syn::Lit::Str(lit),
                ..
            }) => Some(lit.value()),
            Meta::Path(_) => None,
            _ => bail_span!(meta, r#"Expect `trace` or `trace = "manual""#),
        };
        Ok(Some(inner))
```